pydantic/_internal/_decorators.py

```diff
@@ -322,16 +324,28 @@ def gather_decorator_functions(cls: type[Any]) -> DecoratorInfos:
             shimmed_func = var_value.shim(func) if var_value.shim is not None else func
             info = var_value.decorator_info
             if isinstance(info, ValidatorDecoratorInfo):
-                res.validator[var_name] = Decorator(var_name, shimmed_func, func, info)
+                res.validator[var_name] = Decorator(cls.__name__, var_name, shimmed_func, func, info)
             elif isinstance(info, FieldValidatorDecoratorInfo):
-                res.field_validator[var_name] = Decorator(var_name, shimmed_func, func, info)
+                res.field_validator[var_name] = Decorator(cls.__name__, var_name, shimmed_func, func, info)
             elif isinstance(info, RootValidatorDecoratorInfo):
-                res.root_validator[var_name] = Decorator(var_name, shimmed_func, func, info)
+                res.root_validator[var_name] = Decorator(cls.__name__, var_name, shimmed_func, func, info)
             elif isinstance(info, FieldSerializerDecoratorInfo):
-                res.field_serializer[var_name] = Decorator(var_name, shimmed_func, func, info)
+                # check whether a serializer function is already registered for fileds
+                for field_ser_dec in res.field_serializer.values():
```

Comment:
could we rename `field_ser_dec` to be a bit quicker to understand? I would suggest `field_serializer_decorator`, it just makes it easier for me to immediately understand what's going on. If you/others disagree, it can stay as is.

Reply:
Yeah, `field_serializer_decorator `  was my initial name but I changed it because I thought its too long :). Changed it!